### PR TITLE
Fix : Picture correction not being saved [SDESK-6735]

### PR DESCRIPTION
**Pull request by devketanpro**

### CODE REVIEW

File: apps/publish/content/common.py [hunk 1]
```diff
@@ -723,7 +723,7 @@ def _set_updates_for_media_items(self, doc, updates):
             return
 
         for key in DEFAULT_SCHEMA.keys():
-            if doc.get(key) and not updates.get(key):
+            if doc.get(key) and doc.get(key) == updates.get(key):
                 updates[key] = doc[key]
```
Comment (petrjasek) [inline apps/publish/content/common.py:727]: this seems to me like it would never change the `updates` if it only sets the values which are already equal

**Comment by petrjasek** [comment]
pls add/update a test case, I'm not sure I understand the fix

**Comment by devketanpro** [comment]
> pls add/update a test case, I'm not sure I understand the fix

ok I will

File: apps/publish/content/common.py [hunk 2]
```diff
@@ -723,7 +723,7 @@ def _set_updates_for_media_items(self, doc, updates):
             return
 
         for key in DEFAULT_SCHEMA.keys():
-            if doc.get(key):
+            if doc.get(key) and updates.get("operation") == "publish":
```
Comment (petrjasek) [inline apps/publish/content/common.py:726]: might be more readable if that condition for publish operation is part of that previous one as it's not really key related so we won't need it in the loop:
```python
if doc.get("type") not in MEDIA_TYPES or updates.get("operation") == "publish":
    return
```

**Review by petrjasek — APPROVED** [review]

**Pull request merged**